internal/lib/stats/metrics.go

```diff
@@ -0,0 +1,445 @@
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_bytes_total",
+				Help:      "Cumulative count of bytes transmitted",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_packets_total",
+				Help:      "Cumulative count of packets transmitted",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_packets_dropped_total",
+				Help:      "Cumulative count of packets dropped while transmitting",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			}, {
+				Name:      "container_network_transmit_errors_total",
+				Help:      "Cumulative count of errors encountered while transmitting",
+				LabelKeys: append(baseLabelKeys, "interface"),
+			},
+		},
+		"oom": {
+			{
+				Name:      "container_oom_events_total",
+				Help:      "Count of out of memory events observed for the container",
+				LabelKeys: baseLabelKeys,
+			},
+		},
+		"processes": {
+			{
+				Name:      "container_processes",
+				Help:      "Number of processes running inside the container.",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_file_descriptors",
+				Help:      "Number of open file descriptors for the container.",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_sockets",
+				Help:      "Number of open sockets for the container.",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_threads_max",
+				Help:      "Maximum number of threads allowed inside the container, infinity if value is zero",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_threads",
+				Help:      "Number of threads running inside the container",
+				LabelKeys: baseLabelKeys,
+			},
+			{
+				Name:      "container_ulimits_soft",
+				Help:      "Soft ulimit values for the container root process. Unlimited if -1, except priority and nice",
+				LabelKeys: append(baseLabelKeys, "ulimit"),
+			},
+		},
+	}
+	return descriptorsMap
+}
+
+func sandboxBaseLabelValues(sb *sandbox.Sandbox) []string {
+	// TODO FIXME: image?
+	return []string{sb.ID(), "POD", ""}
+}
+
+// ComputeSandboxMetrics computes the metrics for both pod and container sandbox.
+func ComputeSandboxMetrics(sb *sandbox.Sandbox, c *oci.Container, stats []*ContainerStats, metric string, sm *SandboxMetrics) []*types.Metric {
+	values := append(sandboxBaseLabelValues(sb), metric)
+	metrics := make([]*types.Metric, 0, len(stats))
+
+	for _, m := range stats {
+		metricValues := m.valueFunc()
+		if len(metricValues) == 0 {
+			// No metrics to process for this ContainerMetrics, move to the next one
+			continue
+		}
+
+		for _, v := range metricValues {
+			existingMetric := findExistingSandboxMetric(sm.metric, m.desc.Name, values, v.labels)
+			if existingMetric != nil {
+				existingMetric.Value = &types.UInt64Value{Value: v.value}
+			} else {
+				if metric == NetworkMetrics {
```

Comment:
following up on the comment above: I think for now we should keep it simples and just continually replace the slices. We could get fancier in the future if we deem it necessary, but I think we're prone to issues doing so